php/libraries/SessionID.php

```diff
@@ -50,9 +50,9 @@ public function getType(): string
      */
     protected function validate(string $value): bool
     {
-        return (intval($value) > 0)
-            && (strlen($value) <= self::SESSIONID_MAX_LENGTH)
-            && (is_integer($value));
+        return intval($value) > 0
+            && strlen($value) <= self::SESSIONID_MAX_LENGTH
+            && ctype_digit($value);
```

Comment:
what was wrong with is_integer?

Reply:
> is_int() returns true if the argument is an integer type, ctype_digit() takes a string argument and returns true if all the characters in the string are digits.

The value is passed as a string so `is_integer()` always returns false

https://stackoverflow.com/questions/236406/is-there-a-difference-between-is-int-and-ctype-digit